spec/model/implementation/flex-search-validation.spec.ts

```diff
@@ -120,7 +113,25 @@ describe('FlexSearch', () => {
                 someString: String @flexSearch
             }
         `,
-            `At least one field on type "HandlingUnitInfo" should be marked with "includeInSearch".`,
+            `At least one field on type "HandlingUnitInfo" should be marked with "includeInSearch".`
+        );
+    });
+
+    it('rejects recursion of includeInSearch', () => {
+        const validationResult = validate(`
+            type Delivery @rootEntity(flexSearch: true) {
+                handlingUnits: [HandlingUnit] @flexSearch(includeInSearch: true)
+            }
+            type HandlingUnit @childEntity{
+                handlingUnitNumber: String @flexSearch(includeInSearch: true)
+                childHandlingUnits: [HandlingUnit] @flexSearch(includeInSearch: true)
+            }
+        `);
+        expect(validationResult.hasErrors()).to.be.true;
+        // The error appears on "Delivery.handlingUnits" and "HandlingUnit.childHandlingUnits"
+        expect(validationResult.getErrors().length, validationResult.toString()).to.equal(2);
+        expect(validationResult.getErrors()[0].message, validationResult.toString()).to.equal(
+            `"includeInSearch" cannot be used here because it would cause a recursion.`
         );
     });
```

Comment:
you could add a test that tests self-recursion (like "innerHandlingUnits" in type HandlingUnit of type HandlingUnit), these sometimes behave a little different depending on the recursion check (I think yours works though)

Reply:
Isn't that test already for a self-recursion? I added a test that includes a recursion with another type in between, is that what you meant?

Reply:
uhrgs yes of course :D